packages/aws-cdk-lib/aws-codebuild/lib/project.ts

```diff
@@ -1334,13 +1334,33 @@ export class Project extends ProjectBase {
     const hasEnvironmentVars = Object.keys(vars).length > 0;
 
     const errors = this.buildImage.validate(env);
+
+    // Lambda compute does not support timeoutInMinutes property
+    if (this.isLambdaComputeType(this.buildImage)) {
+      if (props.timeout) {
+        errors.push('Cannot specify timeoutInMinutes for lambda compute');
+      }
+      if (props.queuedTimeout) {
+        errors.push('Cannot specify queuedTimeoutInMinutes for lambda compute');
+      }
+      if (props.cache) {
+        errors.push('Cannot specify cache for lambda compute');
+      }
+      if (props.badge) {
+        errors.push('Cannot specify badgeEnabled for lambda compute');
+      }
+    }
+
     if (errors.length > 0) {
       throw new Error('Invalid CodeBuild environment: ' + errors.join('\n'));
     }
 
-    const imagePullPrincipalType = this.buildImage.imagePullPrincipalType === ImagePullPrincipalType.CODEBUILD
-      ? ImagePullPrincipalType.CODEBUILD
-      : ImagePullPrincipalType.SERVICE_ROLE;
+    // For Lambda compute, specifying imagePullPrincipalType is not supported
+    const imagePullPrincipalType = this.isLambdaComputeType(this.buildImage)
```

Comment:
Is this needed?
If CFN does not deploy it should be validated in `validateLambdaBuildImage` otherwise I think we can just ignore the value.

Reply:
In the original implementation, `SERVICE_ROLE` is specified as the default value even if `imagePullPrincipalType` is not specified.
https://github.com/aws/aws-cdk/blob/0178cd39fca2dfe0ae1872b823e6a79dc91a1c78/packages/aws-cdk-lib/aws-codebuild/lib/project.ts#L1341

```ts
    const imagePullPrincipalType = this.buildImage.imagePullPrincipalType === ImagePullPrincipalType.CODEBUILD
        ? ImagePullPrincipalType.CODEBUILD
        : ImagePullPrincipalType.SERVICE_ROLE;

    return {
      imagePullCredentialsType: imagePullPrincipalType,
    };
```

Therefore, I think some kind of modification other than validation is necessary.
Any advice on a better implementation would be appreciated!

Reply:
I think that `imagePullPrincipalType`, `secretsManagerCredentials`, and `repository` will just be ignored if a Lambda-compute image is passed (I might be wrong on this).
If this is the case we could provide some feedback with `Annotations.addWarningV2()` to notify the users that those properties will not be used.
I don't think we should set `imagePullPrincipalType` to `undefined` if `this.isLambdaBuildImage(this.buildImage)` because it would either unset an already ignored value or mask a wrong property specification that should be validated.

Reply:
> I think that imagePullPrincipalType, secretsManagerCredentials, and repository will just be ignored if a Lambda-compute image is passed (I might be wrong on this).

I agree with you.
`imagePullPrincipalType`, `secretsManagerCredentials`, and `repository` are all specified within `IBuildImage`, but the Lambda Build Image doesn't have these properties. Therefore, users generally cannot explicitly specify these values.

However, when it comes to `imagePullPrincipalType`, even if these values are not specified within the Lambda Build Image, the `Project.renderEnvironment` method defaults to setting `imagePullPrincipalType` to `SERVICE_ROLE`. This causes deployment to result in an error message `Cannot specify imagePullPrincipalType for lambda compute`.

So, I think some modifications is needed in lines like the following within the `renderEnvironment` method for the deployment to work correctly.
https://github.com/aws/aws-cdk/blob/14662072692e5555cdfe44d9cfa5d9d66e03c7bf/packages/aws-cdk-lib/aws-codebuild/lib/project.ts#L1341
https://github.com/aws/aws-cdk/blob/14662072692e5555cdfe44d9cfa5d9d66e03c7bf/packages/aws-cdk-lib/aws-codebuild/lib/project.ts#L1364